InteractionBase/src/handlers/GenericHandler.cpp

```diff
@@ -510,6 +514,33 @@ void GenericHandler::keyPressEvent(Visualization::Item *target, QKeyEvent *event
 
 		scene->scheduleUpdate();
 	}
+	else if (event->key() == Qt::Key_F12)
+	{
+		event->accept();
+		if (SelectionAtCursorItem::isVisible())
+			SelectionAtCursorItem::hide();
+		else
+		{
+			QVector<Model::Node*> nodes;
```

Comment:
I don't think you should do any of the things that you're doing here. Just call SelectionAtCursorItem::show, and let it figure out all the rest. No need to pollute the generic handler with that.